src/components/MessageList/MessageList.js

```diff
@@ -210,7 +211,10 @@ const MessageList = (props) => {
   }
 
   const messagesWithDates = insertDates(messages);
-  const messageGroupStyles = getGroupStyles(messagesWithDates, noGroupByUser);
+  const messageGroupStyles = getGroupStyles({
+    messagesWithDates,
+    noGroupByUser,
+  });
   const readData = getReadStates(messagesWithDates, read);
 
   const messageList = messagesWithDates
```

Comment:
Can we extract it to separate hook and export it? Something like `useMessageList`. List of messages will be hydrated with dates, groupStyles and readstate inside this hook.

```js
const messageList = useMessageList();
```

Reply:
Common pattern that I have seen among users is that, they build their own UI components, but they still want to use these kind of utilities ... so they endup copying large chunks of code. With re-usable hooks, we can solve this problem for them

Reply:
Done